src/server/hotkey_collector.cpp

```diff
@@ -105,13 +117,30 @@ void hotkey_collector::on_start_detect(dsn::replication::detect_hotkey_response
 
 void hotkey_collector::on_stop_detect(dsn::replication::detect_hotkey_response &resp)
 {
-    _state.store(hotkey_collector_state::STOPPED);
-    _internal_collector.reset();
+    terminate_colletor();
     resp.err = dsn::ERR_OK;
     std::string hint =
         fmt::format("{} hotkey stopped, cache cleared", dsn::enum_to_string(_hotkey_type));
     ddebug_replica(hint);
 }
 
+void hotkey_collector::terminate_colletor()
+{
+    _state.store(hotkey_collector_state::STOPPED);
+    _internal_collector.reset();
+    _collector_start_time = 0;
+}
+
+void hotkey_collector::terminate_by_timeout()
+{
+    if (dsn_now_s() == 0)
```

Comment:
How can it happen?

Reply:
https://github.com/apache/incubator-pegasus/pull/625/files/df197b1e14bc8b60f9ba1d60079005edaab7d166#diff-7d14c5784befda2ae449c643a1ec2dd2b9988db5fc72325f48e855b6a566e02eR40
https://github.com/apache/incubator-pegasus/pull/625/files/df197b1e14bc8b60f9ba1d60079005edaab7d166#diff-7d14c5784befda2ae449c643a1ec2dd2b9988db5fc72325f48e855b6a566e02eR131
In `hotkey_collector_state::STOPPED` and `hotkey_collector_state::STOPPED` it should be 0

Reply:
It is a mistake